keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,252 @@
+
+* An Application ID (also known as the AAD Client ID or appID)
+* An authentication key (also known as the shared secret).
+
+The application must present both these values to Azure Active Directory, to get a token. 
+
+#### Create/Get Credentials
+
+Create an application in the Azure Active Directory.
+```Powershell
+az ad sp create-for-rbac -n <application-name> --password <application-password> --skip-assignment
+# If you don't specify a password, one will be created for you.
+```
+
+To authorize the same application to perform secret operarions in your vault, type the following command:
+```Powershell
+az keyvault set-policy --name <keyvault-name> --spn 8f8c4bbd-485b-45fd-98f7-ec6300b7b4ed --secret-permissions <secret-permissions>
+```
+
+#### Create Client
+
+In order to interact with the Azure Key Vault Secrets service you'll need to create an instance of the Secret Client class. To make this possible you'll need the application id and application key of an application in Azure Active Directory authorized with access to key vault.
+
+Once you have the values of the appliication id and application key you can create the secret client:
+
+```Java
+SecretClient client = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+```
+
+or
+
+```Java
+SecretAsyncClient client = SecretAsyncClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+```
+
+## Key concepts
+
+### Secret
+
+A secret is the fundamental resource within an Azure KeyVault. In its simplest form it is a name and a value. However, there are additional properties such as the modifiable content type and tags fields that allow the value to be interpreted or associated in different ways.
+The expires property of a Secret provides a way to specify a UTC time at which it will no loger be active. The notBefore property of a Secret allows to specify a UTC time after which secret will be active.
+
+### Secret Client
+
+The client performs the interactions with the Azure Key Vault service, getting, setting, updating, deleting, and listing secrets. An asynchronous, `SecretAsyncClient`, and synchronous, `SecretClient`, client exists in the SDK allowing for selection of a client based on an application's use case.
+
+## Sync API Examples
+
+The following sections provide several code snippets covering some of the most common Azure Key Vault Secret Service tasks, including:
+- [Create a Secret](#create-a-Secret)
+- [Retrieve a Secret](#retrieve-a-Secret)
+- [Update an existing Secret](#update-an-existing-Secret)
+- [Delete a Secret](#delete-a-Secret)
+
+### Create a Secret
+
+Create a Secret to be stored in the Azure Key Vault. There are two ways to store a Secret:
```

Comment:
the wording is somehow confusing. Like I was expecting setSecret and something else as there are 2 ways ....
Maybe removing that last sentence and just leave the `setSecret` explanation ...

Reply:
Removed 2 ways sentence.